### PR TITLE
v1.10 backports 2021-10-04

### PR DESCRIPTION
**Pull request by glibsm**
* #16969 -- docs: clarify language on libceph and kernel 5.8 in kubeproxy-free GSG (@bluikko)
 * #16974 -- daemon: Make L2 neighbor discovery configurable. (@bjhaid)
 * #17341 -- proxy: Expose cachedSelectorREEntry type (@nebril)
 * #17489 -- test: bump coredns version to 1.7.0 (@aanm)
 * #17469 -- Add neighbor discovery behavior docs to kubeproxy-free. (@bjhaid)
 * #17154 -- datapath: Add a new option to skip socket lb when in pod ns (@brb)
 * #17492 -- daemon, ipam, option: Introduce ability to bypass IP availability error (@christarazi)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 16969 16974 17341 17489 17469 17154 17492; do contrib/backporting/set-labels.py $pr done 1.10; done
```
or with
```
$ make add-label branch=v1.10 issues=16969,16974,17341,17489,17469,17154,17492
```

### CODE REVIEW

File: daemon/cmd/daemon_main.go [hunk 1]
```diff
@@ -234,6 +234,9 @@ func init() {
 	flags.Duration(option.ARPPingRefreshPeriod, 5*time.Minute, "Period for remote node ARP entry refresh (set 0 to disable)")
 	option.BindEnv(option.ARPPingRefreshPeriod)
 
+	flags.Bool(option.EnableL2NeighDiscovery, true, "Enables L2 neighbor discovery used by kube-proxy-replacement and IPsec")
+	option.BindEnv(option.EnableL2NeighDiscovery)
+
```
Comment (glibsm) [inline daemon/cmd/daemon_main.go:239]: Don't think this should have gone in, need to double check original PR

**Review by aanm — APPROVED** [review]
LGTM for my commit

**Review by christarazi — APPROVED** [review]
Thanks

**Review by brb — APPROVED** [review]
My changes LGTM, thanks.

**Review by nebril — APPROVED** [review]

**Comment by glibsm** [comment]
/test-backport-1.10

**Pull request merged**